nemo/collections/nlp/models/dialogue/dialogue_zero_shot_slot_filling_model.py

```diff
@@ -0,0 +1,783 @@
+
+        # mention class prediction based on ground truth slots, at mention level
+        mention_loss_mask = mention_loss_mask > 0.5
+        slot_similarity_preds = torch.argmax(dot_product_score_log_softmax, axis=-1)
+        self.slot_similarity_classification_report.update(
+            slot_similarity_preds[mention_loss_mask], mention_labels[mention_loss_mask]
+        )
+
+        # mention class prediction based on predicted bio slots, at token level
+        subtokens_mask = subtokens_mask > 0.5
+        predicted_slot_similarity_preds = torch.argmax(predicted_dot_product_score_log_softmax, axis=-1)
+        predicted_slot_class_batch = DialogueZeroShotSlotFillingModel.align_mention_to_tokens(
+            bio_slot_preds, predicted_slot_similarity_preds
+        )
+        self.overall_slot_classification_report.update(
+            predicted_slot_class_batch[subtokens_mask], slot_labels[subtokens_mask]
+        )
+
+        return {
+            'val_loss': val_loss,
+            'bio_slot_tp': self.bio_slot_classification_report.tp,
+            'bio_slot_fn': self.bio_slot_classification_report.fn,
+            'bio_slot_fp': self.bio_slot_classification_report.fp,
+            'slot_similarity_tp': self.slot_similarity_classification_report.tp,
+            'slot_similarity_fn': self.slot_similarity_classification_report.fn,
+            'slot_similarity_fp': self.slot_similarity_classification_report.fp,
+            'overall_slot_tp': self.overall_slot_classification_report.tp,
+            'overall_slot_fn': self.overall_slot_classification_report.fn,
+            'overall_slot_fp': self.overall_slot_classification_report.fp,
+            'bio_slot_preds': bio_slot_preds,
+            'bio_slot_labels': bio_slot_labels,
+            'slot_similarity_preds': slot_similarity_preds,
+            'overall_slot_preds': predicted_slot_class_batch,
+            'slot_labels': slot_labels,
+            'input': input_ids,
+            'subtokens_mask': subtokens_mask,
+        }
+
+    @staticmethod
+    def align_mention_to_tokens(bio_labels, mention_labels):
+        """
+        Align the mentions label to token level based on bio labels.
+        
+        Args:
+            bio_labels: (batch_size*max_token_length) [[1, 0, 0, 1, 2], [1, 2, 2, 0, 0]]
+            mention_labels: [[2, 4, 0, 0, 0], [52, 0, 0, 0, 0]]
+        
+        Returns:
+            predicted_slot_class: [[2, 0, 0, 4, 4], [52, 52, 52, 0, 0]]
+        """
+        predicted_slot_class = []
+        for one_bio_labels, one_mention_labels in zip(bio_labels, mention_labels):
+            start_and_end = DialogueZeroShotSlotFillingModel.get_start_and_end_for_bio(one_bio_labels)
+            one_predicted_slot_class = torch.zeros(len(one_mention_labels))
+
+            for idx, one_start_and_end in enumerate(start_and_end):
+                start, inclusive_end = one_start_and_end
+                for idy in range(start, inclusive_end + 1):
```

Comment:
If you want to adopt the new implementation, make sure to use the exclusive_end as below, so no need for inclusive_end +1 in  the `range` call